test/e2e/set/scylladbmanagertask/multidatacenter/scylladbmanagertask_scylladbcluster_globalmanager_object_storage.go

```diff
@@ -0,0 +1,556 @@
+		defer taskDeletionCtxCancel()
+		err = framework.WaitForObjectDeletion(
+			taskDeletionCtx,
+			nsClient.DynamicClient(),
+			scyllav1alpha1.GroupVersion.WithResource("scylladbmanagertasks"),
+			smt.Namespace,
+			smt.Name,
+			pointer.Ptr(smt.UID),
+		)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Verifying that the task has been removed from the global ScyllaDB Manager state")
+		// 'GetTask' is broken and does not return an error after the task has been deleted.
+		// XRef: https://github.com/scylladb/scylla-manager/issues/4400
+		tasks, err := managerClient.ListTasks(ctx, sourceManagerClusterID, managerclient.BackupTask, false, "", "")
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		o.Expect(slices.ContainsFunc(tasks.TaskListItemSlice, func(t *managerclient.TaskListItem) bool {
+			return t.ID == managerTaskID.String()
+		})).To(o.BeFalse())
+
+		// Close the existing session to avoid polluting the logs.
+		di.Close()
+
+		framework.By("Deleting the source ScyllaDBCluster")
+		err = nsClient.ScyllaClient().ScyllaV1alpha1().ScyllaDBClusters(ns.Name).Delete(
+			ctx,
+			sourceSC.Name,
+			metav1.DeleteOptions{
+				PropagationPolicy: pointer.Ptr(metav1.DeletePropagationForeground),
+				Preconditions: &metav1.Preconditions{
+					UID: &sourceSC.UID,
+				},
+			},
+		)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		framework.By("Waiting for the source ScyllaDBCluster to be deleted")
+		err = framework.WaitForObjectDeletion(
+			ctx,
+			nsClient.DynamicClient(),
+			scyllav1alpha1.GroupVersion.WithResource("scylladbclusters"),
+			sourceSC.Namespace,
+			sourceSC.Name,
+			pointer.Ptr(sourceSC.UID),
+		)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		targetSC := f.GetDefaultScyllaDBCluster(rkcMap)
+		metav1.SetMetaDataLabel(&targetSC.ObjectMeta, naming.GlobalScyllaDBManagerRegistrationLabel, naming.LabelValueTrue)
+
+		for dcIdx, dcSpec := range targetSC.Spec.Datacenters {
+			workerClusterKey := dcSpec.Name
+			objectStorageSettings := f.GetObjectStorageSettingsForWorkerCluster(workerClusterKey)
+
+			o.Expect(objectStorageSettings.Type()).To(o.BeElementOf(framework.ObjectStorageTypeGCS, framework.ObjectStorageTypeS3))
+			switch objectStorageSettings.Type() {
+			case framework.ObjectStorageTypeGCS:
+				gcServiceAccountKey := objectStorageSettings.GCSServiceAccountKey()
+				o.Expect(gcServiceAccountKey).NotTo(o.BeEmpty())
+
+				o.Expect(err).NotTo(o.HaveOccurred())
+				targetSC.Spec.Datacenters[dcIdx] = *setUpGCSCredentialsForScyllaDBClusterDatacenter(ctx, nsClient.KubeClient().CoreV1(), &dcSpec, ns.Name, gcServiceAccountKey)
+
+			case framework.ObjectStorageTypeS3:
+				s3CredentialsFile := objectStorageSettings.S3CredentialsFile()
+				o.Expect(s3CredentialsFile).NotTo(o.BeEmpty())
+
+				o.Expect(err).NotTo(o.HaveOccurred())
+				targetSC.Spec.Datacenters[dcIdx] = *setUpS3CredentialsForScyllaDBClusterDatacenter(ctx, nsClient.KubeClient().CoreV1(), &dcSpec, ns.Name, s3CredentialsFile)
+
+			}
+		}
```

Comment:
This block is duplicated, it could be a helper function

Reply:
done